File: ruby/fizzbuzz/src/application.rb
# frozen_string_literal: true
# rbs_inline: enabled

# @rbs num: Integer
# @rbs return: String
def fizzbuzz_in_if(num)
  if (num % 3).zero? && (num % 5).zero?
    'FizzBuzz'
  elsif (num % 3).zero?
    'Fizz'
  elsif (num % 5).zero?
    'Buzz'
  else
    num.to_s
  end
end

# @rbs num: Integer
# @rbs return: String
def fizzbuzz_in_if_and_ternary(num)
  if (num % 3).zero?
    (num % 5).zero? ? 'FizzBuzz' : 'Fizz'
  else
    (num % 5).zero? ? 'Buzz' : num.to_s
  end
end

# @rbs num: Integer
# @rbs return: String
def fizzbuzz_in_ternary(num)
  if (num % 3).zero?
    (num % 5).zero? ? 'FizzBuzz' : 'Fizz'
  else
    (num % 5).zero? ? 'Buzz' : num.to_s
  end
end

File: python/fizzbuzz/src/application.py
def fizzbuzz_in_if(num: int) -> str:
    if num % 3 == 0 and num % 5 == 0:
        result = 'FizzBuzz'
    elif num % 3 == 0:
        result = 'Fizz'
    elif num % 5 == 0:
        result = 'Buzz'
    else:
        result = str(num)
    return result


def fizzbuzz_in_if_and_ternary(num: int) -> str:
    if num % 3 == 0:
        return 'FizzBuzz' if num % 5 == 0 else 'Fizz'
    else:
        return 'Buzz' if num % 5 == 0 else str(num)


def fizzbuzz_in_ternary(num: int) -> str:
    return ('FizzBuzz' if num % 5 == 0 else 'Fizz') if num % 3 == 0 \
        else ('Buzz' if num % 5 == 0 else str(num))
